spring-session/src/main/java/org/springframework/session/data/hazelcast/config/annotation/web/http/EnableHazelcastHttpSession.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.hazelcast.config.annotation.web.http;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.session.config.annotation.web.http.EnableSpringHttpSession;
+
+/**
+ * Add this annotation to a {@code @Configuration} class to expose the
+ * SessionRepositoryFilter as a bean named "springSessionRepositoryFilter" and
+ * backed by Hazelcast. In order to leverage the annotation, a single {@link HazelcastInstance}
+ * must be provided. For example:
+ * <pre>
+ * <code>
+ * {@literal @Configuration}
+ * {@literal @EnableHazelcastHttpSession}
+ * public class HazelcastHttpSessionConfig {
+ *
+ *     {@literal @Bean}
+ *     {@literal @ConditionalOnMissingBean}
+ *     public HazelcastInstance hazelcastInstance(Config hazelcastConfig) {
+ *         if (StringUtils.hasText(hazelcastConfig.getInstanceName())) {
+ *             return Hazelcast.getOrCreateHazelcastInstance(hazelcastConfig);
+ *         }
+ *         return Hazelcast.newHazelcastInstance(hazelcastConfig);
+ *     }
+ *
+ * }
+ * </code>
+ * </pre>
+ *
+ * More advanced configurations can extend {@link HazelcastHttpSessionConfiguration} instead.
+ * 
+ * @author Tommy Ludwig
+ * @since 1.1
+ * @see EnableSpringHttpSession
+ */
+@Retention(value=java.lang.annotation.RetentionPolicy.RUNTIME)
+@Target(value={java.lang.annotation.ElementType.TYPE})
+@Documented
+@Import(HazelcastHttpSessionConfiguration.class)
+@Configuration
+public @interface EnableHazelcastHttpSession {
+	int maxInactiveIntervalInSeconds() default 1800;
```

Comment:
@rwinch I'm not sure how we would be able to read the provided configuration. The user could put the configuration in any xml file and include that configuration. Alternatively, they could configure it in Java, reading the values from a properties file, perhaps.

@manderson23 The problem is, as far as I can tell, we have no way to know if the user configured `maxIdleSeconds` as 0 or if they did not configure it and it is just 0 by default. In case they configured it as 0, I think we would want to respect that and not overwrite it.

I'm leaning more towards having to change the default attribute value to 0 to match to default of `maxIdleSeconds`. Then the user can configure it themselves and we will only override if a value is passed to `maxInactiveIntervalInSeconds`. Though this is different than the Redis configuration, I believe. So we have to choose to play it safe and avoid overwriting the user's configuration, or maintain consistency with the equivalent Redis configuration.